### DISCUSSION TITLE
Vector search with filter / match

### DISCUSSION THREAD

**Discussion by ahirel** [FalkorDB/FalkorDB#633] [Ideas]
Hi everyone,

I am currently trying to implement a system that takes for arguments two embedding vectors and returns the K paths that have the highest combines similarities. Currently, the only thing i manage to do is find K nodes for start and K nodes for end, and then try to find matches between them.

```
CALL db.idx.vector.queryNodes('Model', 'vect', 5, vecf32($embeddings_1)) YIELD node as start, score as score_start
CALL db.idx.vector.queryNodes('Model', 'vect', 5, vecf32($embeddings_2)) YIELD node as end, score as score_end

MATCH p = (start)-[*]->(end) WHERE score_start * score_end < 1 

RETURN
	nodes(p) as nodes,
	relationships(p) as relationships
```

However, this seems to be quite different from what i want to do in the end, and not very optimized, since the path that has the best "combined similaraty" is not necessarily a path that has its start in top 5 similarities and end in top 5 similarities.

And with only the queryNodes procedure, it does not seem possible to get the K nearest neighbors given a condition, all that is possible is to get the K nearest neighbors, then filter so the only way seems to go wide with the queryNodes and then hope that enough items match the condition.

Would it be possible, also, for a given fixed node in the graph to have a procedure that only searches among nodes accessible by paths coming from that start node?

I don't know if i am asking for impossible things, or what is the roadmap for the vector search features, or maybe I am just missing something in how the procedure works.  Does someone have an idea on how i could solve this problem?

Thank you very much

**Comment by swilly22** [comment]
Couple of follow up questions:
1. What do you mean by "combines similarities"?
2. In the query you've provided `start_score_2` isn't defined.

Can you please provide a concrete example to what it is you're trying to achieve ?
Thank you!
